### DISCUSSION TITLE
macro test in neovim

### DISCUSSION THREAD

**Discussion by SteveLauC** [clangd/clangd#1230] [Q&A] [answered]
Thanks for this great language server.
And sorry for my ambiguous question title, I don't know how to precisely name it.

Let me give an example:
```c
#include <stdio.h>

int main(void)
{
#ifdef __WIN64
	printf("windows\n");
#endif

#ifdef __linux__
	printf("linux");
#endif
	return 0;
}
```
If I edit the above code under Linux, then the syntax highlight for `windows` snippet should be turned off, as `__WIN64` is not defined on Linux.

And `clangd` do have this feature under `VS Code`:
![VS Code with clangd](https://github.com/SteveLauC/pic/blob/main/Screenshot%20from%202022-07-28%2012-06-29.png)

But in `Neovim`, this feature is disabled by default:
![Neovim with clangd](https://github.com/SteveLauC/pic/blob/main/Screenshot%20from%202022-07-28%2012-06-59.png)

So I am wondering is there any way we can enable this under `Neovim` by editing configuration or something?

**Comment by HighCommander4** [comment, marked as answer]
clangd sends a semantic token with token kind `comment` for the inactive region. The reason why that doesn't affect neovim's rendering needs to be investigated on the neovim side. Some possible reasons I can think of:

  * maybe neovim doesn't have any style associated with the `comment` semantic token kind?
  * maybe neovim doesn't support multi-line tokens?
  * maybe there's an interaction between neovim's client-side highlighting and clangd's server side highlighting that results in this rendering?

**Reply by SteveLauC** [reply]
Thanks for your reply. I kinda think that this stuff is not a part of the Language Server Protocol, cause rust-analyzer (another language server I am using) does have this feature in Neovim. Instead of turning syntax highlight off of that code snippet, rust-analyzer treats it as a warning: [screenshot](https://github.com/SteveLauC/pic/blob/main/Screenshot%20from%202022-07-30%2011-31-35.png).

Thanks for your reply, maybe I should go to [vscode-clangd](https://github.com/clangd/vscode-clangd) community to try to find out how they implement this:)

**Reply by SteveLauC** [reply]
You are right, `Neovim` currently does not support semantic tokens, but it will [land](https://github.com/neovim/neovim/pull/15723) in the near future.

Thank you for answering my question:)
